docs/standard/native-interop/best-practices.md

```diff
@@ -190,6 +190,12 @@ handle.Free();
 
 Don't forget that `GCHandle` needs to be explicitly freed to avoid memory leaks.
 
+### Advanced managed wrapper management
+
+Interop abstractions that rely upon managed object lifetime can be particularly difficult. There are broadly three approaches defined [here](../garbage-collection/unmanaged.md). Lifetime management for types that represent higher-level abstractions for native resources, that is "wrappers", should always consider deriving from `SafeHandle` first, implementing the [`IDisposable` pattern](../garbage-collection/implementing-dispose.md), and using Finalizers only as a last resort.
```

Comment:
I think we should have one place in the docs that goes into depth with trickly very advanced topics like object resurrection, and the other places should just have a short shallow discussion and link to the in-depth doc.

I think the linked  ../garbage-collection/implementing-dispose.md is probably the best place for the in-depth content. If you agree, we can mention this in the DO/AVOID bullet points at the top of this doc and just link to the in-depth content from those.

DO use SafeHandles to managed lifetime of objects that encapsulate unmanaged resources
AVOID finalizers  to managed lifetimes of objects that encapsulate unmanaged resources

Reply:
I re-read through the `implementing-dispose.md` and it already calls out this scenario. This reaffirms this suggestion as the most appropriate. Thanks!